app/utils/room.js

```diff
@@ -18,19 +18,35 @@ export const capitalize = (s) => {
 	return s.charAt(0).toUpperCase() + s.slice(1);
 };
 
-export const formatDate = date => moment(date).calendar(null, {
-	lastDay: `[${ I18n.t('Yesterday') }]`,
-	sameDay: 'LT',
-	lastWeek: 'dddd',
-	sameElse: 'L'
-});
-
-export const formatDateThreads = date => moment(date).calendar(null, {
-	sameDay: 'LT',
-	lastDay: `[${ I18n.t('Yesterday') }] LT`,
-	lastWeek: 'dddd LT',
-	sameElse: 'LL'
-});
+export const formatDate = (date) => {
+	const now = moment();
+	const momentDate = moment(date);
+	const diff = momentDate.diff(now, 'days', true);
+
+	// If it is more than one week ago
+	if (diff < -6) {
+		let format = momentDate.format('ll');
+
+		// If it is same year
+		if (now.year() === momentDate.year()) {
+			format = format.replace(momentDate.format('YYYY'), ''); // Remove year
+		}
+
+		return format
+			.replace(/[рг]\./, '') 	// Remove year letter from RU/UK locales
+			.replace(/de/g, '') 		// Remove preposition from PT
```

Comment:
We can't force remove `de` from here as we don't know how it's going to behave on all languages (it can be part of a month)

Reply:
Then what’s your suggestion?
Remove “de” preposition from Portuguese dates was a requirement to do dates more short.
If we don’t remove “de”, the same year date in Portuguese will be something like `19 de nov de`.